main.js

```diff
@@ -1,37 +1,41 @@
 'use strict';
 
-var spawn = require('child_process').spawn
-var defined = require('defined')
-var Ndsamples = require('ndsamples')
-var getDataType = require('dtype')
-var bufferToTypedArray = require('buffer-to-typed-array')
-var rangeFit = require('range-fit')
-var intmin = require('compute-intmin')
-var intmax = require('compute-intmax')
-var pull = require('pull-stream')
-var toPull = require('stream-to-pull-stream')
+var os = require('os');
+var spawn = require('child_process').spawn;
+var defined = require('defined');
+var Ndsamples = require('ndsamples');
+var getDataType = require('dtype');
+var bufferToTypedArray = require('buffer-to-typed-array');
+var rangeFit = require('range-fit');
+var intmin = require('compute-intmin');
+var intmax = require('compute-intmax');
+var pull = require('pull-stream');
+var toPull = require('stream-to-pull-stream');
 
 module.exports = readAudio
 
 function readAudio (opts, onAbort) {
+  var ps;
   opts = defaultOpts(opts)
 
   // get derived opts
   opts = deriveOpts(opts)
-  
-  // run sox process
-  var ps = spawn(
-    opts.soxPath,
-    [
-      '--buffer', opts.buffer,
-      '--bits', opts.bits,
-      '--channels', opts.channels,
-      '--encoding', opts.encoding,
-      '--rate', opts.rate,
-      opts.inFile,
-      '-p'
-    ]
-  )
+
+  // run process
+  if (os.platform() === 'darwin') {
+      ps = spawn(
+        opts.soxPath,
+        [ '--buffer', opts.buffer, '--bits', opts.bits, '--channels', opts.channels,
+          '--encoding', opts.encoding, '--rate', opts.rate, opts.inFile, '-p'
+        ]
+      )
+  } else {
+      ps = spawn(
```

Comment:
what if instead of `os.platform() !== 'darwin'`, we check that `arecord` path exists to run this code path?